ipatests/test_integration/test_user_permissions.py

```diff
@@ -93,3 +93,41 @@ def test_stageuser_show_as_alternate_admin(self):
         # the field Kerberos Keys available must contain True
         result = self.master.run_command(['ipa', 'stageuser-show', stageuser])
         assert 'Kerberos keys available: True' in result.stdout_text
+
+
+class TestInstallClientNoAdmin(IntegrationTest):
+
+    def test_installclient_as_user_admin(self):
+        """ipa-client-install should not use hardcoded admin for principal
+        Related to : https://pagure.io/freeipa/issue/5406
+        """
+        tasks.install_master(self.master)
+        tasks.kinit_admin(self.master)
+        username = 'testuser1'
+        password = self.master.config.admin_password
+        password_confirmation = "%s\n%s\n" % (password,
+                                              password)
+
+        self.master.run_command(['ipa', 'user-add', username,
+                                 '--first', username,
+                                 '--last', username,
+                                 '--password'],
+                                stdin_text=password_confirmation)
+
+        runcmd1 = ['ipa', 'role-add', 'useradmin']
+        self.master.run_command(runcmd1)
+        runcmd2 = ['ipa', 'role-add-privilege', 'useradmin',
+                   '--privileges="Host Enrollment"']
+        self.master.run_command(runcmd2)
+        runcmd3 = ['ipa', 'role-add-member', 'useradmin',
+                   '--users=' + username]
+        self.master.run_command(runcmd3)
+        cmd = ['ipa-client-install', '-U',
+               '--domain', self.client.domain.name,
+               '--realm', self.client.domain.realm,
+               '-p', username,
+               '-w', self.master.config.admin_password,
+               '--server', self.master.hostname]
+        self.client.run_command(cmd)
+        msg = "getent passwd %s@%s" % (username, self.client.domain.name)
+        assert msg in paths.IPACLIENT_INSTALL_LOG
```

Comment:
The log file needs to be opened/read, with something like
```
install_log = self.clients[0].get_file_contents(paths.IPACLIENT_INSTALL_LOG, encoding='utf-8')
assert msg in install_log
```
otherwise the assert is testing if the msg string is contained in the *filename* (as opposed to the file content).